src/WCSimWCAddDarkNoise.cc

```diff
@@ -0,0 +1,254 @@
+#include "WCSimWCAddDarkNoise.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+#include "G4LogicalVolumeStore.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+#include <utility>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCADDDARKNOISE_VERBOSE
+//#define WCSIMWCADDDARKNOISE_VERBOSE
+#endif
+
+WCSimWCAddDarkNoise::WCSimWCAddDarkNoise(G4String name,
+                                   WCSimDetectorConstruction* myDetector)
+  :G4VDigitizerModule(name)
+{
+  this->myDetector = myDetector;
+  DigiHitMap.clear();
```

Comment:
Actually, looking through it a bit more, it's confusing that the variable doesn't get set if not called in the macro. In the other messenger classes in WCSim this is the case (WCSimDetectorMessenger for example), but I see in your WCSimDarkRateMessenger class that you have set the "omittable" Boolean for each parameter to true, which should set the value to the default, for example: 
```C++
SetFrequency->SetParameterName("DarkRate",true);
```
Anyways, I agree that the defaults shouldn't be in two places. I also think that since the parameters are not set by default in the other messenger classes, having the explicit call to set the default could avoid confusion for users later (it is very clear that even if the parameter is not in the macro, the value will be set).